### PR TITLE
Implicit struct_pack

### PR DESCRIPTION
**Pull request by lnkuiper**
Implements #3558

### CODE REVIEW

**Comment by lnkuiper** [comment]
Some tests are failing with `Error : package ‘DBI’ required by ‘duckdb’ could not be found`

**Comment by hannes** [comment]
The issue is unrelated, I'm working on it

**Review by hawkfish — APPROVED** [review]

**Comment by lnkuiper** [comment]
More unrelated CI failures, GH Actions does not want to play nice

File: test/sql/binder/test_implicit_struct_pack.test [hunk 1]
```diff
@@ -0,0 +1,33 @@
+# name: test/sql/binder/test_implicit_struct_pack.test
+# description: Test implicit struct pack when selecting table alias
+# group: [binder]
+
+statement ok
```
Comment (Mytherin) [inline test/sql/binder/test_implicit_struct_pack.test:5]: `PRAGMA enable_verification`

**Review by Mytherin — COMMENTED** [review]
Thanks for the PR! Looks good. Some comments:

**Comment by lnkuiper** [comment]
Thank you for the feedback! I've implemented it, let's wait for CI

**Comment by Mytherin** [comment]
Thanks!

**Pull request merged**